fedimint-core/src/db/mod.rs

```diff
@@ -831,11 +874,77 @@ macro_rules! push_db_key_items {
     };
 }
 
+/// MigrationMap is a BTreeMap that maps DatabaseVersions to async functions.
+/// These functions are expected to "migrate" the database from the keyed
+/// DatabaseVersion to DatabaseVersion + 1.
+pub type MigrationMap<'a> = BTreeMap<
+    DatabaseVersion,
+    for<'b> fn(
+        &'b mut DatabaseTransaction<'a>,
+    ) -> Pin<Box<dyn futures::Future<Output = anyhow::Result<()>> + Send + 'b>>,
```

Comment:
nit: I think we can use a `BoxFuture<'b, T>` here  [docuementation link](https://docs.rs/futures/latest/futures/future/type.BoxFuture.html)
makes it shorter :) I wanted to go through the code base and replace all these patterns anyway at some point.

Reply:
`semgrep` would probably be able to detect it. :)

Reply:
:+1:  added to my todo